qiime_studio/api/jobs.py

```diff
@@ -147,8 +134,8 @@ def callback(future):
             job = JOBS[job_id]
 
             if results is not None:
-                for result, path in zip(results, outputs.values()):
-                    result.save(path)
+                for key, path in outputs.items():
+                    getattr(results, key).save(path)
                 job['outputs'] = {k: v.uuid for k, v in zip(outputs, results)}
```

Comment:
Sorry just noticed this line as well. It suffers the same problem. But you can probably use `_fields` property. @jairideout is there a correct way to get the key-value map from the `Results` object?